src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -591,3 +646,288 @@ RouterExecutorEnd(QueryDesc *queryDesc)
 	queryDesc->estate = NULL;
 	queryDesc->totaltime = NULL;
 }
+
+/*
+ * Walks each TargetEntry of the query, evaluates sub-expressions without Vars.
+ */
+void ExecuteFunctions(Query *query)
+{
+	CmdType commandType = query->commandType;
+	ListCell *targetEntryCell = NULL;
+	Node *modifiedNode = NULL;
+
+	/* DELETE has no targetList */
+	if (!(commandType == CMD_INSERT) && !(commandType == CMD_UPDATE))
+	{
+		return;
+	}
+
+	foreach(targetEntryCell, query->targetList)
```

Comment:
Do you mean something like:

```
UPDATE tbl SET a = 10 WHERE part_col = 'val' AND last_updated > now() - interval '1 hour'
DELETE FROM tbl WHERE part_col = 'val' AND last_updated > now() - interval '1 hour'
```

I think the same rules apply, STABLE functions can be safely evaluated so long as they don't include a Var (VOLATILE functions in WHERE clauses are also evaluated once for every row). But that doesn't sound hard to also handle.